dbt/include/teradata/macros/adapters.sql

```diff
@@ -118,8 +118,27 @@
 {% endmacro %}
 
 {% macro teradata__get_columns_in_relation(relation) -%}
-    {% set use_qvci = var("use_qvci", "false") | as_bool %}
-     {{ log("use_qvci set to : " ~ use_qvci) }}
+    {% call statement('check_table_or_view', fetch_result=True) %}
+        SELECT TableKind FROM DBC.TablesV WHERE DatabaseName = '{{ relation.schema }}' AND TableName = '{{ relation.identifier }}'
+    {% endcall %}
+
+    {% set table_kind = load_result('check_table_or_view').table.columns['TableKind'].values()[0] %}
+    {{ log("Table Kind : " ~ table_kind) }}
+
+    {%- if table_kind == 'V ' -%}
+        {{ log("Table Kind is V") }}
+        {% set temp_relation_for_view = relation.identifier ~ '_viw_tbl' %}
+        {{ log("Table temp_relation_for_view : " ~ temp_relation_for_view) }}
```

Comment:
Is this logging really required ?
This might be confusing for end user, they don't have anything to do with this.

Reply:
Not required, will remove that